packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,367 @@
+
+    const hasValidationErrors = validateFiles(droppedFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(droppedFiles);
+      onDrop?.({ name, fileList: allFiles });
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const inputFiles = Array.from(event.target.files ?? []);
+    const allFiles = selectedFiles.length > 0 ? [...selectedFiles, ...inputFiles] : inputFiles;
+
+    const hasValidationErrors = validateFiles(inputFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(inputFiles);
+      onChange?.({ name, fileList: allFiles });
+    }
+  };
+
+  return (
+    <BaseBox
+      display="flex"
+      flexDirection="column"
+      {...metaAttribute({ name: MetaConstants.FileUpload, testID })}
+      {...getStyledProps(styledProps)}
+      width="100%"
+    >
+      <BaseBox
+        display="flex"
+        flexDirection={isLabelLeftPositioned ? 'row' : 'column'}
+        alignItems={isLabelLeftPositioned ? 'center' : undefined}
+        position="relative"
+        width="100%"
+      >
+        {label ? (
+          <FormLabel
+            as="span"
+            necessityIndicator={necessityIndicator}
+            position={labelPosition}
+            id={labelId}
+            accessibilityText={accessibilityText}
+          >
+            {label}
+          </FormLabel>
+        ) : null}
+
+        <BaseBox display="flex" flexDirection="column" width="100%">
+          <SelectorLabel
+            componentName={MetaConstants.FileUploadLabel}
+            inputProps={{}}
+            style={{
+              cursor: isDisabled ? 'not-allowed' : 'pointer',
+              ...(isOneFileSelectedWithSingleUpload ? screenReaderStyles : {}),
+            }}
+          >
+            <BaseBox
+              display="flex"
+              flexDirection="column"
+              width="100%"
+              marginBottom={willRenderHintText ? 'spacing.0' : 'spacing.5'}
+            >
+              <StyledFileUploadWrapper
+                isDisabled={isDisabled}
+                isActive={isActive}
+                display="flex"
+                flexDirection="row"
+                justifyContent="center"
+                alignItems="center"
+                borderRadius="medium"
+                borderWidth="thin"
+                borderColor={
+                  isDisabled
+                    ? fileUploadColorTokens.border.disabled
+                    : fileUploadColorTokens.border.default
+                }
+                onDragOver={handleDragOver}
+                onDragLeave={handleDragLeave}
+                onDrop={handleDrop}
+                onClick={() => setIsActive(true)}
+              >
+                <Box
+                  display="flex"
+                  justifyContent="center"
+                  alignItems="center"
+                  flexDirection={{ xs: 'column', s: 'row' }}
```

Comment:
```suggestion
                  flexDirection={{ base: 'column', s: 'row' }}
```

Is this expected btw? so we want it to be flex on smaller phones than normal `s` size phones?

Reply:
yes,